src/docs/asciidoc/advancedFeatures.adoc

```diff
@@ -496,191 +499,195 @@ parallel-2 -> 8
-than coming up with the hook yourself:
+The `Hooks` utility class provides a couple of preset hooks. These are alternatives to
+the default behaviors that you can use by calling their corresponding method, rather than
+coming up with the hook yourself:
 
- - `onNextDroppedFail()`: `onNextDropped` used to throw a `Exceptions.failWithCancel()`
- exception. It now defaults to logging the dropped value at DEBUG level. To go back to the
- old default behavior of throwing, use `onNextDroppedFail()`.
+* `onNextDroppedFail()`: `onNextDropped` used to throw a `Exceptions.failWithCancel()`
+exception. It now defaults to logging the dropped value at the DEBUG level. To go back to
+the old default behavior of throwing, use `onNextDroppedFail()`.
 
- - `onOperatorDebug()`: this activates <<debug-activate,debug mode>>. It ties into the
- `onOperatorError` hook, so calling `resetOnOperatorError()` will also reset it. It can be
- independently reset via `resetOnOperatorDebug()` as it uses a specific key internally.
+* `onOperatorDebug()`: This method activates <<debug-activate,debug mode>>. It ties into
+the `onOperatorError` hook, so calling `resetOnOperatorError()` also resets it. It can be
+independently reset via `resetOnOperatorDebug()` as it uses a specific key internally.
 
 [[context]]
 == Adding a Context to a Reactive Sequence
 One of the big technical challenges encountered when switching from an imperative
-programming perspective to a reactive programming mindset lies in how you deal with threading.
+programming perspective to a reactive programming mindset lies in how you deal with
+threading.
 
-Contrary to what you're used to, in reactive programming a `Thread` can (and will) be used
+Contrary to what you might be used to, in reactive programming, a `Thread` can be used
 to process several asynchronous sequences that run roughly at the same time (actually, in
-non-blocking locksteps). The execution can also very easily and very often jump from a
-thread to another.
+non-blocking locksteps). The execution can also easily and often jump from one thread to
+another.
 
-This is especially hard for developers that use features dependent on the threading model
-being more "stable", like `ThreadLocal`. As it allows you to associate data with a
-*thread*, it becomes tricky to use in a reactive context. As a result, libraries that rely
-on `ThreadLocal` will at least introduce new challenges when used with Reactor, if not
-work badly. Using the MDC of Logback to store and log correlation IDs is a prime example
-of such a situation.
+This arrangement is especially hard for developers that use features dependent on the
+threading model being more "stable", such as `ThreadLocal`. As it lets you associate
+data with a *thread*, it becomes tricky to use in a reactive context. As a result,
+libraries that rely on `ThreadLocal` at least introduce new challenges when used with
+Reactor, even if they do not work badly. Using the MDC of Logback to store and log
```

Comment:
"even if they do not work badly": my intent here was to say that at best they'll introduce new challenges, and at worst they will fail / work badly. any suggestion? the reformulation is counter to what I wanted to say (kinda implies it will never break a library)